Resolved issue where`PaginatedCatalogTable` wasn't passing options

nb-spot opened the pull request:
## Hey, I just made a Pull Request!
Resolved an issue where the `PaginatedCatalogTable` was not propagating table options to its child table.

Before: 

`PaginatedCatalogTable` has the props passed to it properly
![image](https://github.com/backstage/backstage/assets/130386160/30fb92fd-fa40-4d83-be1b-71c4a7dddf37)

But it does not pass them down to its child table
![image](https://github.com/backstage/backstage/assets/130386160/d51481fa-8fb4-4ae1-ba5c-cb333877ec10)

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [-] Added or updated documentation
- [-] Tests for new functionality and regression tests for bug fixes
- [-] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

alde reviewed (approved):
LGTM, nice catch

aramissennyeydd commented:
Hey, thanks for this 👋 , I think this is actually the expected behavior. The paginated catalog page has no understanding of page size, you can pass page size on the initial `pagination` props for your `DefaultCatalogPage` or `EntityListProvider` and this will change how many results appear in the table. There's ongoing work to support this in #22958, but that's not guaranteed to be accepted (pinging @vinzscam on this).

This would be a worthwhile change for any of the props that we're not overriding though, something like
```
{{
...options,
pageSize: Number.MAX_VALUE,
// other options
}}
```

nb-spot commented:
> Hey, thanks for this 👋 , I think this is actually the expected behavior. The paginated catalog page has no understanding of page size, you can pass page size on the initial `pagination` props for your `DefaultCatalogPage` or `EntityListProvider` and this will change how many results appear in the table. There's ongoing work to support this in #22958, but that's not guaranteed to be accepted (pinging @vinzscam on this).
> 
> This would be a worthwhile change for any of the props that we're not overriding though, something like
> 
> ```
> {{
> ...options,
> pageSize: Number.MAX_VALUE,
> // other options
> }}
> ```

Hey! You are absolutely right, @vinzscam confirmed elsewhere that the pagination portions are expected. Updated the PR to allow the other settings through and added a comment.

aramissennyeydd reviewed (approved):
Thanks, looks good!
